core/deployment/src/main/java/io/quarkus/deployment/CodeGenerator.java

```diff
@@ -169,6 +171,47 @@ public static boolean trigger(ClassLoader deploymentClassLoader,
         });
     }
 
+    public static Config getConfig(ApplicationModel appModel, LaunchMode launchMode, Properties buildSystemProps,
+            QuarkusClassLoader deploymentClassLoader) throws CodeGenException {
+        // Config instance that is returned by this method should be as close to the one built in the ExtensionLoader as possible
```

Comment:
@radcortez I tried to keep it as close to the ExtensionLoader one as I could

Reply:
Sounds good!